src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/yflows/YFlowPingSpec.groovy

```diff
@@ -54,8 +61,58 @@ class YFlowPingSpec extends HealthCheckSpecification {
         yFlow && yFlowHelper.deleteYFlow(yFlow.YFlowId)
     }
 
+    @Tidy
+    @Tags([LOW_PRIORITY])
+    def "Able to ping y-flow when one of subflows is one-switch one (#5019)"() {
+        given: "y-flow which has one-switch subflow"
+        def switchTriplet = topologyHelper.getSwitchTriplets(true, true).find {
+            it.shared == it.ep1 && it.shared != it.ep2
+        }
+        assumeTrue(switchTriplet != null, "These cases cannot be covered on given topology:")
+        def yFlow = yFlowHelper.addYFlow(yFlowHelper.randomYFlow(switchTriplet))
+
+        and: "expected ping response"
+        def multiSwitchSubFlowId = yFlow.getSubFlows()
+                .find { it.getEndpoint().getSwitchId().equals(switchTriplet.getEp2().getDpId()) }
+                .getFlowId()
+        def expectedResponseSubflowPart = UniSubFlowPingPayload.builder()
+                .pingSuccess(true)
+                .build()
+        def expectedResponse = YFlowPingResult.builder()
+                .yFlowId(yFlow.getYFlowId())
+                .pingSuccess(false)
+                .error("One sub_flow is one-switch flow")
+                .subFlows([SubFlowPingPayload.builder()
+                                   .flowId(multiSwitchSubFlowId)
+                                   .forward(expectedResponseSubflowPart)
+                                   .reverse(expectedResponseSubflowPart)
+                                   .build()])
+                .build()
+
+        when: "ping y-flow"
+        def response = northboundV2.pingYFlow(yFlow.getYFlowId(), new YFlowPingPayload(2000))
+        response = 'replace unpredictable latency values from ping response'(response)
+
+        then: "y-flow ping is not successful, but one of subflows ping is successful"
+        response == expectedResponse
+
+        cleanup:
+        Wrappers.silent{yFlowHelper.deleteYFlow(yFlow.getYFlowId())}
+    }
+
     def getPacketCountOfVlanPingRule(SwitchId switchId) {
         return northbound.getSwitchRules(switchId).flowEntries
                 .findAll { it.cookie == Cookie.VERIFICATION_UNICAST_RULE_COOKIE }[0].packetCount
     }
+
+    def 'replace unpredictable latency values from ping response'(YFlowPingResult originalResponse) {
```

Comment:
The yflow happy case is not being tested right? what about the classic flows?

Reply:
I tested bugfix only. Pinging flows/yflows are tested many times e.g. when create/update of flows are verified.

Reply:
I said this because I was wondering how to handle the unpredictability of latencies in ping responses in other tests

Reply:
When the PingResponse test model is introduced, all the tests with ping should be refactored to use the one.